### DISCUSSION TITLE
“Listen” to config.json file changes

### DISCUSSION THREAD

**Discussion by derrick-tao** [jdholtz/auto-southwest-check-in#79] [Ideas]
Is it possible for the script to automatically listen to the config.json file changes? Basically I want to make this configuration file public on a web server so that others can input their flights and the script to automatically take on the changes. Currently, I can use a cronjob to restart the script every midnight but maybe there’s a better way?

**Comment by jdholtz** [comment]
There are a few Linux tools you can use (I’m not sure about Windows) to accomplish this. 

You can use Python to do this by using [watchdog](https://pypi.org/project/watchdog/). This might work on Windows too. 

You can also use [inotify](https://github.com/inotify-tools/inotify-tools/wiki) to restart the script when the config file changes. 

Implementing this feature within the script would be pretty difficult because all of the processes in the script need to be updated, not just the main one. 

I’m interested to know what you end up doing to accomplish this. Good luck!
